code/modules/surgery/organs/augments_arms.dm

```diff
@@ -837,3 +837,95 @@
 	if(emp_proof)
 		return
 	muscle_implant.emp_act(severity, owner)
+
+/obj/item/melee/mantis_blade
+	name = "mantis blade"
+	desc = "A blade designed to be hidden just beneath the skin. The brain is directly linked to this bad boy, allowing it to spring into action."
+	icon = 'icons/obj/weapons/melee.dmi'
+	lefthand_file = 'icons/mob/inhands/implants_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/implants_righthand.dmi'
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	tool_behaviour = TOOL_CROWBAR // to pry open doors
+	toolspeed = 0.5
+	usesound = 'sound/items/crowbar.ogg'
+	new_attack_chain = TRUE
+	var/double_attack = TRUE
+	var/double_attack_cd = 1.5 // seconds, so every second attack
+	sharp = TRUE
+	w_class = WEIGHT_CLASS_BULKY
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "lacerated", "ripped", "diced", "cut")
+
+/obj/item/melee/mantis_blade/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot == ITEM_SLOT_LEFT_HAND)
+		transform = matrix(-1, 0, 0, 0, 1, 0)
+	else
+		transform = null
+
+// make double attack if blades in both hands and not on CD
+/obj/item/melee/mantis_blade/attack(mob/living/target, mob/living/user, params, second_attack = FALSE)
```

Comment:
Could you move the actual attacking logic out to a proc other than `attack()`, and handle the double-attack stuff there? I'm not a huge fan of how it's being given an argument like this, as well as how you're calling it recursively outside of the attack chain.

Reply:
Looks good, thanks! I just didn't want attack being recursive.